### PR TITLE
L2-905: Improve Neuron Id Display

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Improve the display of sns neuron id.

# Changes

* New component ShortenMiddleEllipsis that shortens and adds tooltip with all the text.
* New format util that shortens text.
* Use new ShortenMiddleEllipsis in SNSNeuronCard.
* Add the copy icon and functoinality in the SNS Neuron Card for the id.

# Tests

* Test new format util.
* Fix broken SNSNeuronCard test.

### CODE REVIEW

**Comment by lmuntaner** [comment]
New UI:

Screenshot 2022-08-04 at 17 41 01
Screenshot 2022-08-04 at 17 43 39
Screenshot 2022-08-04 at 17 45 54

I also improved the alignment of the Copy icon and account identifier:

Screenshot 2022-08-04 at 17 40 03
Screenshot 2022-08-04 at 17 40 30

File: frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte [hunk 1]
```diff
@@ -15,6 +15,8 @@
   import NeuronCardContainer from "../neurons/NeuronCardContainer.svelte";
   import NeuronStateInfo from "../neurons/NeuronStateInfo.svelte";
   import NeuronStateRemainingTime from "../neurons/NeuronStateRemainingTime.svelte";
+  import Copy from "../ui/Copy.svelte";
+  import ShortenMiddleEllipsis from "../ui/ShortenMiddleEllipsis.svelte";
```
Comment (peterpeterparker) [inline frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte:19]: maybe a dumb idea, what do you think of naming this new ui component `<Hash />`?

File: frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte [hunk 2]
```diff
@@ -38,8 +40,14 @@
...
-  <div slot="start" data-tid="sns-neuron-card-title">
-    <h3 data-tid="neuron-id">{neuronId}</h3>
+  <div class="identifier" slot="start" data-tid="sns-neuron-card-title">
+    <ShortenMiddleEllipsis
+      id="neuron-id"
+      tagName="h3"
+      testId="neuron-id"
+      text={neuronId}
+    />
+    <Copy value={neuronId} />
```
Comment (peterpeterparker) [inline frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte:50]: are we going to use `ShortenMiddleEllipsis` without `Copy`?
if not, should `Copy` be moved in `ShortenMiddleEllipsis`?

**Comment by peterpeterparker** [comment]
Minor questions, looks cool 👍

File: frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte:19]: I like that it's shorter, sure.

File: frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte:50]: I can add it inside and like in `Identifier` we can pass a prop to hide it

File: frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/components/sns-neurons/SnsNeuronCard.svelte]: Coolio

**Review by peterpeterparker — APPROVED** [review]
Looks cool! Thanks

**Pull request merged**